includes/class-wc-gateway-amazon-payments-advanced.php

```diff
@@ -788,7 +788,7 @@ public function maybe_handle_apa_action() {
 	 */
 	public function get_checkout_session_id() {
 		$session_key = $this->get_checkout_session_key();
-		return WC()->session->get( $session_key, false );
+		return ! empty( WC()->session ) ? WC()->session->get( $session_key, false ) : false;
```

Comment:
I am not sure, for this one. I have doubts that setting a variable to WC()->session, while session is not set could produce a warning. I think since we are using it only twice here and there are no initializations processes involved, it would be ok to keep it this way.

Of course, let me know if you disagree and ill make the change.

Reply:
I see, let's stick to your approach then.